packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -153,64 +153,49 @@ class Dart2JSTarget extends Target {
 
   @override
   Future<void> build(Environment environment) async {
-    final String dart2jsOptimization = environment.defines[kDart2jsOptimization];
-    final bool csp = environment.defines[kCspMode] == 'true';
     final BuildMode buildMode = getBuildModeForName(environment.defines[kBuildMode]);
-    final String specPath = globals.fs.path.join(
-      globals.artifacts.getArtifactPath(Artifact.flutterWebSdk), 'libraries.json');
-    // TODO(jonahwilliams): support configuration of this file.
-    const String packageFile = '.packages';
-    final File outputKernel = environment.buildDir.childFile('app.dill');
-    final File outputFile = environment.buildDir.childFile('main.dart.js');
-    final List<String> dartDefines = decodeDartDefines(environment.defines, kDartDefines);
-    final List<String> extraFrontEndOptions = decodeDartDefines(environment.defines, kExtraFrontEndOptions);
 
-    // Run the dart2js compilation in two stages, so that icon tree shaking can
-    // parse the kernel file for web builds.
-    final ProcessResult kernelResult = await globals.processManager.run(<String>[
+    final List<String> initCommand = <String>[
```

Comment:
since this isn't a function any more I would name it something like "sharedArgs"

Reply:
How about sharedCommandOptions?

Reply:
Sounds good!